sympy/stats/joint_rv_types.py

```diff
@@ -103,7 +104,9 @@ def pdf(self, *args):
     def marginal_distribution(self, indices, sym):
         sym = ImmutableMatrix([Indexed(sym, i) for i in indices])
         _mu, _sigma = self.mu, self.sigma
-        k = len(self.mu)
+        k = self.mu.shape[0]
+        if not k.is_Integer:
+            raise ValueError('k must be of type int.')
```

Comment:
I have raised this error since we are using `for` loop.

Reply:
`k` may be a SymPy integer.

Anyway, you shouldn't raise exceptions. We have handlers for unevaluated expressions.